scripts/bootstrap/bootstrap-haskell.ps1

```diff
@@ -660,10 +676,22 @@ if (!($DontAdjustBashRc)) {
   $AdjustBashRcExport = 'export BOOTSTRAP_HASKELL_ADJUST_BASHRC=1 ;'
 }
 
+if (!($Msys2Env)) {
+   $SupportedArchitectures = 'AMD64', 'x86'
+  if ($env:PROCESSOR_ARCHITECTURE -eq 'x86') {
+	  $Msys2EnvExport = ('export GHCUP_MSYS2_ENV=MINGW32 ;')
+  } elseif ($env:PROCESSOR_ARCHITECTURE -eq 'AMD64') {
+	  $Msys2EnvExport = ('export GHCUP_MSYS2_ENV=MINGW64 ;')
+  }
+} else {
+  $Msys2EnvExport = ('export GHCUP_MSYS2_ENV={0} ;' -f $Msys2Env)
+  $null = [Environment]::SetEnvironmentVariable("GHCUP_MSYS2_ENV", $Msys2Env, [System.EnvironmentVariableTarget]::User)

```

Comment:
Would be helpful with some code comments regarding this entire block, starting with `if (!($Msys2Env)) {`.

For example, why is `GHCUP_MSYS2_ENV` set only if `Msys2Env` is set while `Msys2EnvExport` is set regardless?